src/XMakeBuildEngine/Evaluation/Evaluator.cs

```diff
@@ -1811,22 +1811,12 @@ private void DecorateItemsWithMetadataFromProjectItemElement(ProjectItemElement
                     include.Add(itemElement.Include);
                     ItemsAndMetadataPair itemsAndMetadataFromInclude = ExpressionShredder.GetReferencedItemNamesAndMetadata(include);
 
-                    if (itemsAndMetadataFromInclude.Items != null && itemsAndMetadataFromInclude.Items.Count > 0)
-                    {
-                        needToProcessItemsIndividually = true;
-                    }
-                    else
-                    {
-                        // If there is bare built-in metadata, we must always run items individually, as that almost
-                        // always differs between items.
-
-                        // UNDONE: When batching is implemented for real, we need to make sure that
-                        // item definition metadata is included in all metadata operations during evaluation
-                        if (itemsAndMetadataFound.Metadata.Values.Count > 0)
-                        {
-                            needToProcessItemsIndividually = true;
-                        }
-                    }
+                    // If there is bare built-in metadata, we must always run items individually, as that almost
```

Comment:
I'd vote to just leave this code as is, until we have to fix a bug in it. Then, we can refactor the whole thing.